### PR TITLE
Update Rubocop to 0.72 & merge rubocop-chef

### PR DESCRIPTION
**Pull request by tas50**
This merges in the work that @coderanger and I did years ago to get rubocop ready to replace foodcritic. It's mildly updated from the original codebase since rubocop's APIs have changed a bit.

### CODE REVIEW

File: lib/rubocop/chef/inject.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    module Inject
+      def self.defaults!
+        path = CONFIG_DEFAULT.to_s
+        hash = ConfigLoader.send(:load_yaml_configuration, path)
+        config = Config.new(hash, path)
+        puts "configuration from #{path}" if ConfigLoader.debug?
+        config = ConfigLoader.merge_with_default(config, path)
+        ConfigLoader.instance_variable_set(:@default_configuration, config)
+      end
+    end
```
Comment (lamont-granquist) [inline lib/rubocop/chef/inject.rb]: is this duplicated with whatever else in cookstyle that does config injection already?

Comment (tas50) [reply lib/rubocop/chef/inject.rb]: yeah it's totally duplicate. There's a note about that in the PR. I need to rip all this double injection out.

File: config/cookstyle.yml [hunk 2]
```diff
@@ -3,6 +3,47 @@ AllCops:
   Exclude:
     - vendor/**/*
     - Guardfile
+  ChefAttributes:
```
Comment (tas50) [inline config/cookstyle.yml]: I'm not actually sure if we need any of these. We're not actively using them

**Comment by tyler-ball** [comment]
Discussing this in code review, updating the Rubocop engine will likely break lots of users. The update fixes a lot of bugs so would like start correctly alerting on issues it used to miss. The engine update would also apply to chef-style so would likely break our internal pipelines though.

But in the interest of moving forward and correctly fixing these bugs through an engine update we could merge this PR if we changed all the errors to warnings. It will break some people, but also lets them `cookstyle -a` to fix the warnings they are seeing. It won't automatically fix all the issues they get but is a start.

The last thing to consider is that we just did a major version bump of ChefDK/WS. It would have been nice to get the engine update in for that major version bump but we could get it in a minor version bump and that means we don't have to wait an entire year before getting these fixes in. Not great for early adopters but better than delaying these updates another year.

**Comment by tas50** [comment]
Just a heads up: The tests need to get ported too, but this is the minimal bit we need to ship this and get it in the hands of customers.

**Pull request merged**